src/hooks/__tests__/create.test.js

```diff
@@ -0,0 +1,79 @@
+describe('create hook test', () => {
+  beforeEach(() => {
+    jest.mock('/engine/create-auth-hook', () => ({
+      createAuthHook: jest.fn(),
+      onAuthorized: jest.fn()
+    }));
+  });
+
+  afterEach(() => {
+    jest.resetModules();
+  });
+
+  test('should call create hook proper options', async () => {
+    const { create } = require('../create');
+    const { createAuthHook } = require('../../engine/create-auth-hook');
+    const options = {
+      authenticate: jest.fn(),
+      ruleServiceName: 'authorization-rules',
+      getPrincipalId: jest.fn(),
+      getArguments: jest.fn(),
+      isAuthorized: jest.fn(),
+      onAuthorized: await jest.fn(),
+      attributeLookup: {
+        principalId: jest.fn()
+      }
+    };
+
+    create(options);
+
+    expect(createAuthHook).toHaveBeenCalledWith({
+      authenticate: expect.any(Function),
+      ruleServiceName: 'authorization-rules',
+      getPrincipalId: expect.any(Function),
+      getArguments: expect.any(Function),
+      isAuthorized: expect.any(Function),
+      onAuthorized: expect.any(Function),
+      attributeLookup: {
+        principalId: expect.any(Function)
+      }
+    });
+  });
+
+  test('create onAuthorized should return true', async () => {
+    const { create } = require('../create');
+    const { createAuthHook } = require('../../engine/create-auth-hook');
+    const options = {
+      authenticate: expect.any(Function),
+      ruleServiceName: 'authorization-rules',
+      getPrincipalId: expect.any(Function),
+      getArguments: expect.any(Function),
+      isAuthorized: expect.any(Function),
+      onAuthorized: await jest.fn(),
+      attributeLookup: {
+        principalId: expect.any(Function)
+      }
+    };
+    create(options);
+    return expect(createAuthHook.mock.calls[0][0].onAuthorized({ 'authorized': true, context: 'resolved' })).resolves.toBe('resolved');
```

Comment:
This is awesome. Can we break it onto a few lines though? This is a long statement.